src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/PropertyPages/BuildPropertyPage.xaml

```diff
@@ -196,7 +196,11 @@
     </BoolProperty.DataSource>
     <BoolProperty.Metadata>
       <NameValuePair Name="VisibilityCondition">
-        <NameValuePair.Value>(and (has-net-core-app-version-or-greater "8.0") (has-project-capability "NativeAOTProperties"))</NameValuePair.Value>
+        <NameValuePair.Value>
+          (and
+            (has-project-capability "NativeAOTProperties")
+            (has-evaluated-value "Application" "OutputType" "Library"))
```

Comment:
When we talked about this, there was concern that the project capabilities wouldn't be updated in time for the visibility conditions to see them here.

The reasoning was that the capabilities are updated during evaluation, which happens some time after the project is updated, yet we compute visibility conditions almost immediately.

Is this a problem for this PR? If, for example, you change from .NET 7 to .NET 8 (or vice-versa) does the UI update the visibilities correctly? If not we may need some additional work on the CPS side to refresh visibilities whenever capabilities change.

Reply:
That's correct, yes! I need to open a follow up PR in the CPS side to ensure visibility is refreshed. I'll link to it in this PR when I have it.